### PR TITLE
visualize serverless function in topology

### PR DESCRIPTION
**Pull request by nemesis09**
Story: 
https://issues.redhat.com/browse/ODC-5480

Problem/Description:
Serverless functions should be visually identifiable on topology

Solution:
use distinct icon and background color to distinctly identify serverless function in topology

Screens:
_Topology_
Screenshot from 2021-02-24 17-34-04

_Topology List View_
Screenshot from 2021-02-25 14-49-15

Test Coverage:
Screenshot from 2021-02-24 17-36-14

Browser Conformation:
- [x] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/knative-plugin/src/components/overview/KnativeResourceOverviewPage.tsx [hunk 1]
```diff
@@ -83,12 +84,18 @@ export const KnativeResourceOverviewPage: React.ComponentType<KnativeResourceOve
     );
   }
 
+  const {
+    metadata: { labels },
+  } = item?.obj;
+  const isServerlessFunction = labels && labels['boson.dev/function'];
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/KnativeResourceOverviewPage.tsx:90]:
```suggestion
  const isServerlessFunction = labels?.['boson.dev/function'];
```
and better put label `boson.dev/function` in a const as it has been used multiple places.

File: frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.odc-serverless-function-label {
+    margin-left: var(--pf-global--spacer--md);
+}
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.scss:3]: nit: add a new line here

File: frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Label } from '@patternfly/react-core';
+import * as React from 'react';
+import './ServerlessFunctionLabel.scss';
+
+const ServerlessFunctionLabel: React.FC = () => (
+  <Label variant="filled" className="odc-serverless-function-label" color="purple">
+    Function
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.tsx:7]: localize the string

File: frontend/packages/knative-plugin/src/topology/components/groups/KnativeServiceGroup.tsx [hunk 4]
```diff
@@ -118,6 +119,10 @@ const KnativeServiceGroup: React.FC<KnativeServiceGroupProps> = ({
     width,
     height,
   );
+  const {
+    metadata: { labels },
+  } = getResource(element);
+  const isServerlessFunction = labels && labels['boson.dev/function'];
```
Comment (vikram-raj) [inline frontend/packages/knative-plugin/src/topology/components/groups/KnativeServiceGroup.tsx:125]: Same here
```suggestion
  const isServerlessFunction = labels?.['boson.dev/function'];
```

**Comment by nemesis09** [comment]
@vikram-raj 
I've updated the PR and addressed the reviews. PTAL

File: frontend/packages/knative-plugin/src/components/overview/KnativeResourceOverviewPage.tsx (hunk 1 above)
Comment (nemesis09) [reply frontend/packages/knative-plugin/src/components/overview/KnativeResourceOverviewPage.tsx]: done

File: frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.scss (hunk 2 above)
Comment (nemesis09) [reply frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.scss]: done

File: frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.tsx (hunk 3 above)
Comment (nemesis09) [reply frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.tsx]: done

File: frontend/packages/knative-plugin/src/topology/components/groups/KnativeServiceGroup.tsx (hunk 4 above)
Comment (nemesis09) [reply frontend/packages/knative-plugin/src/topology/components/groups/KnativeServiceGroup.tsx]: done

**Comment by christianvogt** [comment]
@openshift/team-devconsole-ux @serenamarie125 are we still putting the `function` label in the header?

**Comment by invincibleJai** [comment]
Looks good 

cc @vikram-raj

**Comment by vikram-raj** [comment]
@nemesis09 tests are failing because `knative-plugin.json` is not updated. Please run `yarn i18n` and update the PR.

**Comment by nemesis09** [comment]
> @nemesis09 tests are failing because `knative-plugin.json` is not updated. Please run `yarn i18n` and update the PR.

updated

**Comment by vikram-raj** [comment]
image

function icon needs to be updated in the topology list view as well.

**Comment by nemesis09** [comment]
> image
> 
> function icon needs to be updated in the topology list view as well. `nodetest` is knative function.

updated PR addressing the review comment. PTAL

**Review by vikram-raj — COMMENTED** [review]
Verified the changes. and it works as expected

/lgtm

**Pull request merged**